file() was removed in Python 3 in favor of open()

See https://github.com/ros2/rosdistro/pull/231

flake8 testing of https://github.com/ros/rosdistro on Python 3.6.3

$ __flake8 . --count --select=E901,E999,F821,F822,F823 --show-source --statistics__
```
./scripts/add_devel_repo.py:55:21: F821 undefined name 'file'
    yaml.dump(data, file(yaml_file, 'w'), default_flow_style=False)
                    ^
./scripts/add_release_repo.py:28:21: F821 undefined name 'file'
    yaml.dump(data, file(yaml_file, 'w'), default_flow_style=False)
                    ^
./scripts/sort_yaml.py:16:21: F821 undefined name 'file'
    yaml.dump(data, file(yaml_file, 'w'), default_flow_style=False)
                    ^
./scripts/yaml2rosinstall.py:13:21: F821 undefined name 'file'
    yaml.dump(data, file(rosinstall_file, 'w'), default_flow_style=False)
                    ^
4     F821 undefined name 'file'
4
```

--- a/scripts/add_devel_repo.py
+++ b/scripts/add_devel_repo.py
@@ -52,7 +52,8 @@ def add_devel_repository_fuerte(yaml_file, data, name, vcs_type, url, version):
     values['version'] = version
     data['repositories'][name] = values
     sort_yaml_data(data)
-    yaml.dump(data, file(yaml_file, 'w'), default_flow_style=False)
+    with open(yaml_file, 'w') as out_file:
+        yaml.dump(data, out_file, default_flow_style=False)
 
 
 if __name__ == "__main__":

--- a/scripts/add_release_repo.py
+++ b/scripts/add_release_repo.py
@@ -25,7 +25,8 @@ def add_release_repository_fuerte(yaml_file, data, name, url, version):
         'version': version,
     }
     sort_yaml_data(data)
-    yaml.dump(data, file(yaml_file, 'w'), default_flow_style=False)
+    with open(yaml_file, 'w') as out_file:
+        yaml.dump(data, out_file, default_flow_style=False)
 
 
 if __name__ == "__main__":

--- a/scripts/sort_yaml.py
+++ b/scripts/sort_yaml.py
@@ -13,7 +13,8 @@ def sort_yaml(yaml_file):
         print('This script does not support the new rosdistro yaml files', file=sys.stderr)
         sys.exit(1)
     sort_yaml_data(data)
-    yaml.dump(data, file(yaml_file, 'w'), default_flow_style=False)
+    with open(yaml_file, 'w') as out_file:
+        yaml.dump(data, out_file, default_flow_style=False)
 
 
 def sort_yaml_data(data):

--- a/scripts/yaml2rosinstall.py
+++ b/scripts/yaml2rosinstall.py
@@ -10,7 +10,8 @@ import yaml
 def convert_yaml_to_rosinstall(yaml_file, rosinstall_file):
     data = yaml.load(open(yaml_file, 'r'))
     data = convert_yaml_data_to_rosinstall_data(data)
-    yaml.dump(data, file(rosinstall_file, 'w'), default_flow_style=False)
+    with open(rosinstall_file, 'w') as out_file:
+        yaml.dump(data, out_file, default_flow_style=False)
 
 
 def convert_yaml_data_to_rosinstall_data(data):

